modules/core/client/app/init.js

```diff
@@ -10,6 +10,30 @@ angular.module(ApplicationConfiguration.applicationModuleName).config(['$locatio
 	}
 ]);
 
+angular.module(ApplicationConfiguration.applicationModuleName).run(function($rootScope, $state, Authentication) {
+    // Check authentication before changing state
+    $rootScope.$on('$stateChangeStart', function(event, toState, toParams, fromState, fromParams) {
+        if (toState.data && toState.data.roles && toState.data.roles.length > 0) {
+            var allowed = false;
+            toState.data.roles.forEach(function (role) {
+               if (Authentication.user.roles !== undefined && Authentication.user.roles.indexOf(role) !== -1) {
+                   allowed = true;
+                   return true;
+               }
+            });
+
+            if (!allowed) {
+                event.preventDefault();
+                $state.go('authentication.signin', {}, {
```

Comment:
Is there a reason why this sends the user to the signin route? Shouldn't it take them 'home'?